pylot/prediction/utils.py

```diff
@@ -0,0 +1,130 @@
+import math
+import numpy as np
+
+from pylot.utils import Location, Rotation, Transform, Vector2D
+
+def pad_trajectory(trajectory, num_steps):
+    """Take the appropriate number of past steps as specified by num_steps.
+    If we have not seen enough past locations of the obstacle, pad the
+    trajectory with the appropriate number of copies of the earliest
+    location.
+    """
+    num_past_locations = trajectory.shape[0]
+    if num_past_locations < num_steps:
+        initial_copies = np.repeat([np.array(trajectory[0])],
+                                   num_steps - num_past_locations,
+                                   axis=0)
+        trajectory = np.vstack((initial_copies, trajectory))
+    elif num_past_locations > num_steps:
+        trajectory = trajectory[-num_steps:]
+    assert trajectory.shape[0] == num_steps
+    return trajectory
+
+def get_nearby_obstacles(all_obstacles, radius):
+    """Given a list of obstacles with past trajectories in the ego-vehicle's
+       frame of reference, return a list of obstacles that are within a
+       specified radius of the ego-vehicle, sorted by increasing distance
+       from the ego-vehicle.
+    """
+    distances = [
+        v.trajectory[-1].get_angle_and_magnitude(Location())[1]
+        for v in all_obstacles
+    ]
+    sorted_vehicles = [
+        v for v, d in sorted(zip(all_obstacles, distances),
+                             key=lambda pair: pair[1])
+        if d <= radius
+    ]
+
+    return sorted_vehicles
+
+def get_nearby_obstacles_ego_transforms(nearby_obstacles):
```

Comment:
Shouldn't we merge the two get_nearby functions, and move them to ObstacleTrajectoryMessage? IMHO, it's going to make the code simpler and safer because we won't manually deal with two lists.